# Code compiled for Shin et al. 2024
Last amended 10/10/24 by Maddie Urbanek  
Contains code used for analyses in Shin & Urbanek et al. 2024. Code is split into separate Jupyter and R markdown files depending on language used. Command line code (executed on a Mac) for aligning bits and transcriptomes is contained in the first chunk of cells in general_jupyter_analyses.ipynb. Additional descriptions of which analyses are in which files are below and annotated by figure location: 

## general_jupyter_analyses.ipynb
Contains code used for calling barcode bits and transcriptome alignments (to be run in command line, but currently housed in jupyter notebook), barcode diversity analyses (Fig. 1-2, S1), cell type proportion analyses (Fig. 4-5, S6), electrophysiological property analyses (Fig. 5), nuclear and cellular barcode comparisons (Fig. 6), empty droplet analysis (Fig. 7), and descriptive stats for connectivity (Fig. 8).

## general_r_scripts.rmd
Contains code used for transcriptomic and gene expression analyses (DEG, GSEA) (Fig. 4-5, S4, S6), stacked bar plot for empty droplet analysis (Fig. 7), and cell type calling for connectivity analysis (Fig. 8, S7, S8). 

## connectivity_jupyter_analyses.ipynb
Contains code used for calculating Shannon entropy and connectivity analysis (Fig. 1, 2, 8, S7, S8).
